python/cudf/cudf/core/dataframe.py

```diff
@@ -287,7 +288,13 @@ def _getitem_tuple_arg(self, arg):
                         BooleanMask(tmp_arg[0], len(columns_df))
                     )
                 else:
-                    tmp_col_name = str(uuid4())
+                    if columns_df._data.multiindex:
+                        tmp_col_name = (
+                            str(uuid4()),
+                            "" * columns_df._data.nlevels,
```

Comment:
This doesn't seem to actually use the number of levels. `"" * n` is always `""` (repeating an empty string).

Reply:
This should be:
```
tmp_col_name = (str(uuid4()), *("" for _ in range(columns_df._data.nlevels - 1)))
```
You're trying to produce a column index that has the same number of levels as `columns_df` to avoid this warning:

```
In [61]: other.join(df, how="inner")
/home/wence/Documents/src/rapids/cudf/python/cudf/cudf/core/join/join.py:438: FutureWarning: merging between different levels is deprecated and will be removed in a future version. (1 levels on the left, 2 on the right)
  warnings.warn(
```

Reply:
Sorry for the incorrect logic, Thank you both for catching this issue.